tests/functional/send_sms.js

```diff
@@ -134,14 +137,35 @@
      const formattedPhoneNumber =
        countryInfo.format(countryInfo.normalize(testPhoneNumber));
 
-     suite['valid phone number'] = function () {
+     suite['valid phone number, back'] = function () {
        return this.remote
         .then(openPage(SEND_SMS_URL, SELECTOR_SEND_SMS_HEADER))
         .then(type(SELECTOR_SEND_SMS_PHONE_NUMBER, testPhoneNumber))
         .then(click(SELECTOR_SEND_SMS_SUBMIT))
         .then(testElementExists(SELECTOR_SMS_SENT_HEADER))
         .then(testElementTextInclude(SELECTOR_SMS_SENT_TO, formattedPhoneNumber))
-        .then(testElementExists(SELECTOR_MARKETING_LINK));
+        .then(testElementExists(SELECTOR_MARKETING_LINK))
+
+        // user realizes they made a mistake
+        .then(click(SELECTOR_SMS_SENT_BACK))
+        .then(testElementExists(SELECTOR_SEND_SMS_HEADER))
+
+        // original phone number should still be in place
+        .then(testElementValueEquals(SELECTOR_SEND_SMS_PHONE_NUMBER, testPhoneNumber));
+     };
+
+     suite['valid phone number, resend'] = function () {
+       return this.remote
+        .then(openPage(SEND_SMS_URL, SELECTOR_SEND_SMS_HEADER))
+        .then(type(SELECTOR_SEND_SMS_PHONE_NUMBER, testPhoneNumber))
+        .then(click(SELECTOR_SEND_SMS_SUBMIT))
+        .then(testElementExists(SELECTOR_SMS_SENT_HEADER))
+
+        // Give a slight delay or else nexmo throttles the request
+        .sleep(10000)
```

Comment:
Yeah, that's Nexmo though. I was thinking for the functional tests, maybe we could open the content server page with a query parameter that causes the `/sms` page to not actually send the request, but instead just go to the next screen to make sure the UI acts as we expect. Maybe we'd send 1 or 2 requests, 1 to check whether backend phone number validation works, 1 to check whether SMS send is successful.